woocommerce-services.php

```diff
@@ -667,6 +668,58 @@ public function localize_and_enqueue_service_script( $method_id, $instance_id =
 		}
 
 		/**
+		 * Add tracking info (if available) to outgoing emails using the woocommerce_email_after_order_table hook
+		 *
+		 * @param $order
+		 *
+		 * @return string
+		 */
+		public function add_tracking_info_to_emails( $order ) {
+            $id = $order->id;
+
+            if (!$id) return;
+
+            $labels = maybe_unserialize( get_post_meta( $id, 'wc_connect_labels', false ) );
+
+            if ( empty( $labels ) ) return;
+
+            echo '<h2>' . __( 'Tracking', 'woocommerce-services' ) . '</h2>';
+
+            echo '<table class="td" cellspacing="0" cellpadding="6" style="margin-top: 10px; width: 100%; font-family: \'Helvetica Neue\', Helvetica, Roboto, Arial, sans-serif;">';
+            echo '<thead>';
+            echo '<tr>';
+            echo '<th class="td" scope="col">' . __( 'Provider', 'woocommerce-services' ) . '</th>';
+            echo '<th class="td" scope="col">' . __( 'Tracking number', 'woocommerce-services' ) . '</th>';
+            echo '</tr>';
+            echo '</thead>';
+            echo '<tbody>';
+            echo '<tr>';
+
+            foreach ( $labels as &$label ) {
+                $label = $label[0];
+                $carrier = $label['carrier_id'];
+                $tracking = $label['tracking'];
+
+                echo '<td class="td" scope="col">' . strtoupper($carrier) . '</td>';
+
+                switch ( $carrier ) {
+                    case "fedex":
+                        $tracking_url = 'https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=' . $tracking;
+                        break;
+                    case "usps":
+                        $tracking_url = 'https://tools.usps.com/go/TrackConfirmAction.action?tLabels=' . $tracking;
+                        break;
+                }
+
+                echo '<td class="td" scope="col"><a href="' . $tracking_url . '">' . $tracking . '</a></td>';
```

Comment:
Please use `esc_url()` on the tracking url, and `esc_html()` on the tracking number here.

See: https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/#escaping-securing-output